lambdas/log-redaction/src/redact-handler.ts

```diff
@@ -28,19 +41,45 @@ export class RedactHandler implements LambdaInterface {
       const redactLogGroup = logEvents.logGroup + "-redacted";
       const logStream = logEvents.logStream;
 
-      try {
+      const query = await this.dynamodb.send(
+        new QueryCommand({
+          TableName: logStreamTrackingTable,
+          KeyConditionExpression: "logStreamName = :logStream",
+          ExpressionAttributeValues: {
+            ":logStream": {
+              S: logStream,
+            },
+          },
+        })
+      );
+
+      if (query.Count == 0) {
```

Comment:
Please can we have this in a method and saving into dynamodb too. That would make the handler logic shorter and appear more focused on the redaction